packages/uniorg-parse/src/parser.ts

```diff
@@ -1857,6 +1936,20 @@ class Parser {
   private atHeading(): boolean {
     return this.r.lookingAt(/^\*+[ \t]/) !== null;
   }
+
+  /*
+   *  Return begin and end positions from current cursor position + val length
+   */
+  private getCurrentRange(val: string): [number, number] {
```

Comment:
minor: this function seems to be rather unfortunate—most of the usages above misuse it because cursor dependency is not obvious. It seems that it's better to remove this function and inline all the usages (that would probably make the bugs more obvious)